pkg/ext/certs.go

```diff
@@ -236,9 +236,16 @@ func (p *rotatingSNIProvider) Run(stopChan <-chan struct{}) error {
 			}
 		case event, ok := <-watcher.ResultChan():
 			if !ok || event.Type == watch.Error {
-				logrus.Errorf("watcher channel closed: %v", err)
+				switch obj := event.Object.(type) {
```

Comment:
What if we enter this branch because of `!ok`? Then `event` is probably `nil` so we can't log much more than "watcher channel closed". Right now the code would most likely panic.